src/AutoRest.CSharp/Common/AutoRest/Communication/StandaloneGeneratorRunner.cs

```diff
@@ -84,6 +89,29 @@ await foreach (var file in workspace.GetGeneratedFilesAsync())
             }
         }
 
+        private static void DeleteDirectory(string path, string[] keepFiles)
+        {
+            var directoryInfo = new DirectoryInfo(path);
+            foreach (FileInfo file in directoryInfo.GetFiles())
+            {
+                if (keepFiles.Contains(file.Name))
+                {
+                    continue;
+                }
+                file.Delete();
+            }
+
+            foreach (DirectoryInfo directory in directoryInfo.GetDirectories())
+            {
+                DeleteDirectory(directory.FullName, keepFiles);
+            }
+
+            if (directoryInfo.GetFiles().Length == 0 && directoryInfo.GetDirectories().Length == 0)
+            {
+                directoryInfo.Delete();
+            }
```

Comment:
I suggest to use the `directoryInfo.EnumerateFiles()` instead of `GetFiles` especially when you are detecting if the directory is empty.
GetFiles returns an array which is calculated ahead of usage. EnumerateFiles returns an IEnumerable which is calculated when iterated.
If you want to check if the directory is empty, I believe on check `!directoryInfo.EnumerateFileSystemEntries().Any()` could do the same and it is simpler (this I am not quite sure). If this is not the case, I suggest we use `!directoryInfo.EnumerateFiles().Any() && !directoryInfo.EnumerateDirectories().Any()` instead